### PR TITLE
feat: add IPNI itest

### PR DESCRIPTION
**Pull request by LexLuthr**

### CODE REVIEW

File: itests/ipni_publish_test.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+	// Make a deal
+	res, err := f.MakeDummyDeal(dealUuid, carFilepath, rootCid, server.URL+"/"+filepath.Base(carFilepath), false)
+	require.NoError(t, err)
+	require.True(t, res.Result.Accepted)
+	log.Debugw("got response from MarketDummyDeal", "res", spew.Sdump(res))
+
+	time.Sleep(2 * time.Second)
+
+	// Wait for the first deal to be added to a sector and cleaned up so space is made
```
Comment (dirkmc) [inline itests/ipni_publish_test.go:78]:
```suggestion
	// Wait for the first deal to be added to a sector
```

**Review by dirkmc — COMMENTED** [review]
Looks great, nice work 🙌 

I think you may need to add the new file to .circleci/config.yml in order for it to be picked up by the CI test runner (see test-itest-dummydeal)

File: itests/ipni_publish_test.go (hunk 1 above)
Comment (LexLuthr) [reply itests/ipni_publish_test.go:78]: Yeah, I will add those now.

File: itests/ipni_publish_test.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+	found := false
+
+	for _, mh := range mhs {
+		if mh.String() == rootCid.Hash().String() {
+			found = true
+			break
+		}
+	}
+
+	require.True(t, found)
```
Comment (masih) [inline itests/ipni_publish_test.go:108]: Use `require.Contains`?

File: itests/ipni_publish_test.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+	require.NoError(t, err)
+	require.True(t, res.Result.Accepted)
+	log.Debugw("got response from MarketDummyDeal", "res", spew.Sdump(res))
+
+	time.Sleep(2 * time.Second)
+
+	// Wait for the deal to be added to a sector
+	err = f.WaitForDealAddedToSector(dealUuid)
+	require.NoError(t, err)
+	time.Sleep(100 * time.Millisecond)
```
Comment (masih) [inline itests/ipni_publish_test.go:81]: Use `require.Eventually`. Ditto re other `time.Sleep`s.

File: itests/ipni_publish_test.go [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+	fileSize := 200000
+	randomFilepath, err := testutil.CreateRandomFile(tempdir, 5, fileSize)
+	require.NoError(t, err)
+
+	// create a dense carv2 for deal making
+	rootCid, carFilepath, err := testutil.CreateDenseCARv2(tempdir, randomFilepath)
+	require.NoError(t, err)
+
+	// Start a web server to serve the car files
+	log.Debug("starting webserver")
```
Comment (masih) [inline itests/ipni_publish_test.go:61]: Use `t.log`? Ditto in other places.

**Review by masih — COMMENTED** [review]
I would say this is a Unit Test more than an integration test. Valuable nevertheless and thank you for adding this 👍

File: itests/ipni_publish_test.go (hunk 4 above)
Comment (LexLuthr) [reply itests/ipni_publish_test.go:61]: Logging is separate for whole framework in itests

File: itests/ipni_publish_test.go (hunk 3 above)
Comment (LexLuthr) [reply itests/ipni_publish_test.go:81]: I am removing the sleep as we don't really need them.

**Review by dirkmc — APPROVED** [review]
Great work 🚀

**Pull request merged**